multicluster/apis/multicluster/v1alpha1/resourceexport_webhook.go

```diff
@@ -53,8 +53,13 @@ func (r *ResourceExport) Default() {
 	if kindLabelVal, exists := r.Labels[constants.SourceKind]; !exists || kindLabelVal != constants.AntreaClusterNetworkPolicyKind {
 		r.Labels[constants.SourceKind] = constants.AntreaClusterNetworkPolicyKind
 	}
-	if r.DeletionTimestamp.IsZero() && !stringExistsInSlice(r.Finalizers, constants.ResourceExportFinalizer) {
-		r.Finalizers = append(r.Finalizers, constants.ResourceExportFinalizer)
+	// Add domain qualified finalizer for ResourceExports to avoid Kubernetes from reporting errors:
+	//  "prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers"
+	// https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers
+	// Note for ResourceExports created before Antrea v2.2, LegacyResourceExportFinalizer may still be present
+	// and needs to be removed before a ResourceExport is deleted.
+	if r.DeletionTimestamp.IsZero() && !stringExistsInSlice(r.Finalizers, constants.DomainQualifiedResourceExportFinalizer) {
+		r.Finalizers = append(r.Finalizers, constants.DomainQualifiedResourceExportFinalizer)
```

Comment:
Obviously this predates this PR, but I am surprised to see a webhook being in charge of adding the finalizer. Is there a recommendation to do that somewhere? I usually see it done in the controller itself (`Reconcile`). cc @luolanzone

Reply:
Hi @antoninbas I think the Finalizer is not necessarily handled by the reconciler. If I recall it correctly, the main purpose is to stop deleting the ResourceExport during creation, because if it's already converted by the leader controller into ResourceImport, there will be no way to clean up the corresponding info in the ResourceImport when the ResourceExport is deleted by the user.